metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAcl.java

```diff
@@ -96,6 +97,13 @@ public String principal() {
         return principal;
     }
 
+    public KafkaPrincipal kafkaPrincipal() {
```

Comment:
What happens here when there is no colon? I guess type becomes the empty string and name becomes everything?

Should we set type = `User` instead, or throw an exception?